did some more magic

caspahouzer opened the pull request:
- cleanup code
- add joins to get
- add joins to countRows
- add "where id" to update
- add getLastId
- add getLastEntry
- add insertOrReplace

ricardoalcocer commented:
Hi @caspahouzer , thank you for your PR.  Could you please change the sample app so it demonstrates how to use these new features...and also the README...please :) ?

ricardoalcocer commented:
@caspahouzer, there's a couple bugs....general bugs...can't track down when they were introduced.

I'm modifying the sample app so it actually shows the results of the operations on-screen.  There's an error in tableExists - here https://github.com/ricardoalcocer/jssql/blob/master/app/lib/com.alcoapps.dbhelper.js#L424, it's trying to do rs.close, when rs is an array not an actual recordset.

There's also an error in .set (https://github.com/ricardoalcocer/jssql/blob/master/app/lib/com.alcoapps.dbhelper.js#L424), where this is always returning zero.

Are these things working for you?

caspahouzer commented:
Sorry,

there was a bug, because in my actual project I’m not using this tableExists function. Database was already closed in getData.
Fixed that, added also a new function for altering a table. And sent a PR

- Sebastian



> Am 06.05.2015 um 03:12 schrieb Ricardo Alcocer <notifications@github.com>:
> 
> @caspahouzer <https://github.com/caspahouzer>, there's a couple bugs....general bugs...can't track down when they were introduced.
> 
> I'm modifying the sample app so it actually shows the results of the operations on-screen. There's an error in tableExists - here https://github.com/ricardoalcocer/jssql/blob/master/app/lib/com.alcoapps.dbhelper.js#L424 <https://github.com/ricardoalcocer/jssql/blob/master/app/lib/com.alcoapps.dbhelper.js#L424>, it's trying to do rs.close, when rs is an array not an actual recordset.
> 
> There's also an error in .set (https://github.com/ricardoalcocer/jssql/blob/master/app/lib/com.alcoapps.dbhelper.js#L424 <https://github.com/ricardoalcocer/jssql/blob/master/app/lib/com.alcoapps.dbhelper.js#L424>), where this is always returning zero.
> 
> Are these things working for you?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ricardoalcocer/jssql/pull/12#issuecomment-99280029>.
>
